internal/server/singleprocess/poll_application.go

```diff
@@ -184,8 +152,57 @@ func (a *applicationPoll) buildPollJob(
 
 		// TODO define timeout interval based on projects app poll interval
 	}
+	var jobs []*pb.QueueJobRequest
+
+	log.Trace("Determining which target to generate a status report on")
+
+	// Default to poll on the "latest" lifecycle operation, so if there's a
+	// release, queue up a status on release. If the latest is deploy, then queue that.
+	if latestRelease != nil && !latestRelease.Unimplemented {
+		releaseJobCopy, err := copystructure.Copy(baseJob)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to generate a job to poll release status: %s", err)
+		}
+		releaseJob := releaseJobCopy.(*pb.QueueJobRequest)
+		releaseJob.Job.Operation = &pb.Job_StatusReport{
+			StatusReport: &pb.Job_StatusReportOp{
+				Target: &pb.Job_StatusReportOp_Release{
+					Release: latestRelease,
+				},
+			},
+		}
+		// SingletonId so that we only have one poll operation at
+		// any time queued per app/operation.
+		releaseJob.Job.SingletonId = fmt.Sprintf("appl-poll/%s/deployment", app.Name)
```

Comment:
I was pretty sure I found some kind of weird bug in the polling subsystem until I got to here and realized what the deploy and release jobs needed different singleton IDs:

https://github.com/hashicorp/waypoint/blob/c2455052d7e9ad304ae30a155d04c9764caa17ec/internal/server/singleprocess/state/job.go#L1000